Js/fix odometry

Roboterbastler opened the pull request.

Roboterbastler committed:
Added front wheel offset to odometry calculation

Roboterbastler committed:
[tricycle_drive_controller] Add steering angle offset parameter

bochen87 commented on include/tricycle_drive_controller/odometry.h:
```diff
@@ -152,8 +152,10 @@ class Odometry
+     * caster joint axis [m]
      */
-    void setWheelParams(double wheel_base, double wheel_radius);
+    void setWheelParams(double wheel_base, double wheel_radius, double front_wheel_caster_offset);
```
maybe call it "front_wheel_position_offset" or "front_wheel_offset_to_center" ?

bochen87 commented on include/tricycle_drive_controller/tricycle_drive_controller.h:
```diff
@@ -137,6 +137,12 @@ class TricycleDriveController : public controller_interface::MultiInterfaceContr
+    double front_wheel_caster_offset_;
+
+    /// Angular offset of the steering angle
+    double front_wheel_steering_offset_;
```
"front_wheel_angular_offset" or "front_wheel_steering_angle_offset" ?

Roboterbastler replied:
Yeah, maybe that makes it more clear

bochen87 commented on src/tricycle_drive_controller.cpp:
```diff
@@ -264,7 +268,7 @@ TricycleDriveController::update(const ros::Time& time, const ros::Duration& peri
     // COMPUTE AND PUBLISH ODOMETRY
     if (real_hw_) {
-        odometry_.updateFromHWEncoders(front_wheel_cmd.front().getVelocity(), front_wheel_caster_cmd.front().getPosition(), time);
+        odometry_.updateFromHWEncoders(front_wheel_cmd.front().getVelocity(), front_wheel_caster_cmd.front().getPosition() + front_wheel_steering_offset_, time);
```
maybe put the front_wheel_steering_offset variable also in the odometry.cpp? That way all the offset stuff is in one place if you're searching for it.

Roboterbastler replied:
Well, I thought the offset isn't really related to the odometry calculation but just a modification of the values sent to and read from the hardware. I would say it's more consistent to keep it out of the odometry and also use the parameter in only one file...

Roboterbastler committed:
[tricycle_drive_controller] Fix offset not applied in neutral position

Steering angle offset was not applied when steering angle command was 0.
Fixed this bug.

bochen87 reviewed (approved):
maybe put the front_wheel_steering_offset variable also in the odometry.cpp? That way all the offset stuff is in one place if you're searching for it.

Roboterbastler committed:
[tricycle_drive_controller] Refactoring

Roboterbastler commented:
The renamings are done now

Roboterbastler committed:
[tricycle_drive_controller] Refactoring

bochen87 commented:
looks good!

Roboterbastler commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Rework Tricycle Driver](https://trello.com/c/kE8ex12W/362-rework-tricycle-driver)
